src/asmcnc/apps/drywall_cutter_app/tool_material_popup.py

```diff
@@ -0,0 +1,178 @@
+
+        Label:
+            id: tool_label
+            text: 'Tool'
+            font_size: app.get_scaled_sp('15sp')
+            color: hex('#333333')
+            size_hint: (None, None)
+            pos_hint: {'center_x': 0.15, 'center_y': 0.35}
+
+        Choices:
+            id: tool_dropdown
+            size_hint: (0.65, 0.1)
+            pos_hint: {'center_x': 0.55, 'center_y': 0.35}
+        
+        BoxLayout:
+            orientation: 'horizontal'
+            pos_hint: {'x': 0.625, 'y': 0.075}
+            size_hint: (None, None)
+            size: (dp(app.get_scaled_width(189)),dp(app.get_scaled_height(23)))
+            spacing: dp(app.get_scaled_width(20))
+
+            Button:
+                id: cancel_button
+                on_press: root.cancel()
+                size_hint: (None,None)
+                width: dp(app.get_scaled_width(83))
+                height: dp(app.get_scaled_height(23))
+                background_color: [0,0,0,0]
+                BoxLayout:
+                    padding: 0
+                    size: self.parent.size
+                    pos: self.parent.pos
+                    Image:
+                        source: "./asmcnc/apps/drywall_cutter_app/img/cancel_button.png"
+                        center_x: self.parent.center_x
+                        y: self.parent.y
+                        size: self.parent.width, self.parent.height
+                        allow_stretch: True
+
+            Button:
+                id: confirm_button
+                on_press: root.confirm()
+                size_hint: (None,None)
+                width: dp(app.get_scaled_width(86))
+                height: dp(app.get_scaled_height(23))
+                background_color: [0,0,0,0]
+                BoxLayout:
+                    padding: 0
+                    size: self.parent.size
+                    pos: self.parent.pos
+                    Image:
+                        source: "./asmcnc/apps/drywall_cutter_app/img/confirm_button.png"
+                        center_x: self.parent.center_x
+                        y: self.parent.y
+                        size: self.parent.width, self.parent.height
+                        allow_stretch: True
+""")
+
+
+class ToolMaterialPopup(Popup):
+
+    def __init__(self, localization, **kwargs):
+        super(ToolMaterialPopup, self).__init__(**kwargs)
+
+        self.tool_dropdown.bind(text=self.on_tool_change)
+        self.material_dropdown.bind(text=self.on_material_change)
+
+        self.profile_db = App.get_running_app().profile_db
+    def on_tool_change(self, instance, value):
+        if value != '':
+            self.confirm_button.disabled = False
+            Logger.debug("Tool changed to " + value)
+
+    def on_material_change(self, instance, value):
+        Logger.debug("Material changed to " + value)
+        self.tool_dropdown.values = self.profile_db.get_tool_names(self.profile_db.get_material_id(value))
```

Comment:
Does this not cause visual lag? The time complexity is O(m+m*p) I think, would be interested to see how this scales when there's more profiles and materials.

Reply:
Suggest call to get_material_id is replaced by the value being stored against the Id in the dropdown.